tasks/main.yml

```diff
@@ -4,19 +4,13 @@
     register: yum_output
   - debug: var=yum_output
 
-  - name: Reboot Host if packages were updated
-    shell: sleep 2 && shutdown -r now "Ansible updates triggered"
-    async: 1
-    poll: 0
-    ignore_errors: true
-    when: ("Complete!" in yum_output.stdout_lines[-1]) or
-          ("Komplett!" in yum_output.stdout_lines[-1])
+  - name Check if reboot is needed
+    shell:
+      cmd: /usr/bin/needs-restarting -r
```

Comment:
this only works on el7, on el8 command is /usr/bin/dnf needs-restarting -r ; see my comment https://github.com/Tronde/ansible-role-rhel-patchmanagement/issues/12#issuecomment-1009845368